app/src/main/java/com/victor/adasfeed/FeedActivity.kt

```diff
@@ -27,26 +30,55 @@ class FeedActivity : AppCompatActivity() {
     private lateinit var postAdapter: PostAdapter
     private lateinit var storiesAdapter: StoriesAdapter
     private lateinit var headerView: View
+    private lateinit var textName: TextView
+    private lateinit var textNickname: TextView
+    private var user: User? = null
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         Log.d("ciclo de vida", "onCreate")
         Log.d("contexto no onCreate", applicationContext.toString())
         setContentView(R.layout.activity_feed)
 
-        val user = User(
+        textName = findViewById(R.id.textName)
+        textNickname = findViewById(R.id.textNickname)
+
+        var oldUser = User(
             userName = "Andrey Freitas",
             userNickname = "@andreyfreitas",
             imageUser = R.drawable.user1,
             tel = "+55 (11) 123456789"
         )
 
+        val extras = intent.extras
+        var uri = Uri.parse("")
+
+        // envia por intents - putExtra
+        // recebe por Bundle
+        extras?.let { bundle ->
+            user = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+                bundle.getParcelable(EXTRA_KEY, User::class.java)
+            } else {
+                bundle.getParcelable(EXTRA_KEY) as? User
+            }
+            user?.let { safeUser ->
+                textName.text = getString(R.string.profile_name, safeUser.userName)
+                textNickname.text = safeUser.userNickname
+            }
+        }
+
+        if (user == null) {
+            user = oldUser
+        }
+
         val intent = Intent(applicationContext, ProfileActivity::class.java).apply {
             putExtra(EXTRA_KEY, user)
         }
 
         // inicializa views (findViewById())
         initViews()
+        textName.text = user!!.userName
```

Comment:
We should avoid forcing a `null` object to be not-null (`!!`):
To handle this, we could use `let`:

```kotlin
user?.let { safeUser ->
   textName.text = safeUser.userName
   textNickname.text = safeUser.userNickname
}
```